### ISSUE TITLE
Connection failed

### ISSUE THREAD

**Issue by Alexis81** [yknivag/ESP_OTA_GitHub_Showcase#3]
Hello,

I think I did everything right and I still get this error?

Number of CA certs read: 150
Checking for update...
Not proceeding to upgrade: Connection failed

in this part of code ::

bool ESPOTAGitHub::checkUpgrade() {

	_setClock(); // Clock needs to be set to perform certificate checks

    BearSSL::WiFiClientSecure client;
    client.setCertStore(_certStore);

    if (!client.connect(GHOTA_HOST, GHOTA_PORT)) {
        _lastError = "Connection failed";
        return false;
    }

Best Regards

**Comment by yknivag** [comment]
The error suggests a network issue.

Is your ESP8266 properly connected to your WiFi network? And can you connect from that same network to https://api.guthub.com/

**Comment by Alexis81** [comment]
Yes I am well connected to the network, look at the following lines ::

connected with Orange Flybox-75C7, channel 11
dhcp client start...
ip:192.168.1.104,mask:255.255.255.0,gw:192.168.1.1
.
Connected successfull to SSID 'Orange Flybox-75C7'
 - mDNS => OK
Number of CA certs read: 150
Checking for update...
Not proceeding to upgrade: Connection failed

**Comment by yknivag** [comment]
And can you connect from another device on the same network to https://api.github.com/

**Comment by Alexis81** [comment]
I connect well on the same network from my computer

{
  "current_user_url": "https://api.github.com/user",
  "current_user_authorizations_html_url": "https://github.com/settings/connections/applications{/client_id}",
  "authorizations_url": "https://api.github.com/authorizations",
  "code_search_url": "https://api.github.com/search/code?q={query}{&page,per_page,sort,order}",
  "commit_search_url": "https://api.github.com/search/commits?q={query}{&page,per_page,sort,order}",
  "emails_url": "https://api.github.com/user/emails",
  "emojis_url": "https://api.github.com/emojis",
  "events_url": "https://api.github.com/events",
  "feeds_url": "https://api.github.com/feeds",
  "followers_url": "https://api.github.com/user/followers",
  "following_url": "https://api.github.com/user/following{/target}",
  "gists_url": "https://api.github.com/gists{/gist_id}",
  "hub_url": "https://api.github.com/hub",
  "issue_search_url": "https://api.github.com/search/issues?q={query}{&page,per_page,sort,order}",
  "issues_url": "https://api.github.com/issues",
  "keys_url": "https://api.github.com/user/keys",
  "label_search_url": "https://api.github.com/search/labels?q={query}&repository_id={repository_id}{&page,per_page}",
  "notifications_url": "https://api.github.com/notifications",
  "organization_url": "https://api.github.com/orgs/{org}",
  "organization_repositories_url": "https://api.github.com/orgs/{org}/repos{?type,page,per_page,sort}",
  "organization_teams_url": "https://api.github.com/orgs/{org}/teams",
  "public_gists_url": "https://api.github.com/gists/public",
  "rate_limit_url": "https://api.github.com/rate_limit",
  "repository_url": "https://api.github.com/repos/{owner}/{repo}",
  "repository_search_url": "https://api.github.com/search/repositories?q={query}{&page,per_page,sort,order}",
  "current_user_repositories_url": "https://api.github.com/user/repos{?type,page,per_page,sort}",
  "starred_url": "https://api.github.com/user/starred{/owner}{/repo}",
  "starred_gists_url": "https://api.github.com/gists/starred",
  "user_url": "https://api.github.com/users/{user}",
  "user_organizations_url": "https://api.github.com/user/orgs",
  "user_repositories_url": "https://api.github.com/users/{user}/repos{?type,page,per_page,sort}",
  "user_search_url": "https://api.github.com/search/users?q={query}{&page,per_page,sort,order}"
}

**Comment by yknivag** [comment]
Just to confirm is this when you are using the example in this repository to upgrade to the binary attached here? Or have you made your own sketch with your own repository using the library?

Also if you use the "BearSSL_CertStore" example in the Arduino IDE can your ESP8266 connect to "api.github.com"?

Lastly which version of the ESP board definitions are you using?
